packages/smooth_app/lib/pages/product/nutrition_page_loaded.dart

```diff
@@ -83,15 +84,21 @@ class _NutritionPageLoadedState extends State<NutritionPageLoaded> {
       }
       children.add(_addNutrientButton(appLocalizations));
     }
-    children.add(_addCancelSaveButtons(
-      appLocalizations,
-      localDatabase,
-    ));
 
     return Scaffold(
-      appBar: AppBar(title: Text(appLocalizations.nutrition_page_title)),
+      appBar: AppBar(
+        title: Text(appLocalizations.nutrition_page_title),
+        actions: <Widget>[
+          IconButton(
+              onPressed: () => _validateAndSave(localDatabase),
```

Comment:
Let's not  block the PR on this, but although I like the idea I still have two things to put out

Just saving can lead to accidental changes, how about an alert dialog with a confirmation prompt?

We should implement the same logic for pop using the system back button